### ISSUE TITLE
Improve/reconsider Spec compliance of history-traversal task-source and task-queue

### ISSUE THREAD

**Issue by gterzian** [servo/servo#21815]
It appears that the Spec contains two related concepts:
1. A [#session-history-traversal-queue](https://html.spec.whatwg.org/multipage/#session-history-traversal-queue),
2. A [#history-traversal-task-source](https://html.spec.whatwg.org/multipage/webappapis.html#history-traversal-task-source)

The queue under 1 appears to be a concept to be run "in parallel", and currently Servo appears to implement it via `History` sending messages directly to the constellation, which appears correct if only a little implicit in it's implementation.  See [`History.traverse_history`](https://github.com/servo/servo/blob/391d67fa1d3509cc95ae79dd8f5c0dd33624d9a6/components/script/dom/history.rs#L69) and Step 1 of[#traverse-the-history-by-a-delta](https://html.spec.whatwg.org/multipage/#traverse-the-history-by-a-delta).

This queue doesn't seem to have an explicit task source associated with it, like the  other script task queues does.

Therefore, the "history-traversal" task source isn't actually used to queue tasks on that parallel history-traversal queue, instead the task-source is used to queue task on a script event-loop, for example see the document unload steps in Step 5 of #traverse-the-history-by-a-delta. 

Currently, while we do have a `HistoryTraversalTaskSource`, it doesn't seem like we're using it to enqueue the task mentioned at Step 5, and also the task source doesn't actually implement `TaskSource` and appears un-used. 

We could consider implementing TaskSource for it, and use it in Step 5 of #traverse-the-history-by-a-delta

cc @cbrewster @AgustinCB

**Comment by gterzian** [comment]
Related https://github.com/whatwg/html/issues/4054

**Comment by gterzian** [comment]
Closing in favor of https://github.com/servo/servo/issues/10994
